web/src/components/dashboard/SidebarCardFundingNudge.tsx

```diff
@@ -13,16 +29,93 @@ type Props = {
 
 export const SidebarCardFundingNudge = (props: Props): ReactElement => {
   const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [tempProfileData, setTempProfileData] = useState<ProfileData>(createEmptyProfileData());
+
+  const { queueUpdateTaskProgress } = useUpdateTaskProgress();
+  const close = (): void => {};
+  const onSuccess = (): void => {};
   const router = useRouter();
   const { updateQueue, business } = useUserData();
 
-  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+  useMountEffectWhenDefined(() => {
     if (!business) return;
+    setTempProfileData(business.profileData);
+  }, business);
+
+  const updateToUpAndRunningAndCompleteTaxTask = async (): Promise<void> => {
+    if (!business || !updateQueue) return;
     await updateQueue
       ?.queueProfileData({
         operatingPhase: OperatingPhaseId.UP_AND_RUNNING,
       })
       .update();
+
+    const encryptedTaxId = checkEncryptedTaxId({
+      tempProfileDataTaxId: tempProfileData.taxId,
+      businessTaxId: business.profileData.taxId,
+      encryptedTaxId: tempProfileData.encryptedTaxId,
+    });
+
+    try {
+      const businessNameToSubmitToTaxApi = nameToSubmitToTaxApi({
+        tempProfileData,
+        business,
+      });
+
+      if (!businessNameToSubmitToTaxApi) {
+        return;
+      }
+
+      const userDataToSet = await postTaxFilingsOnboarding({
+        taxId: tempProfileData.taxId as string,
+        businessName: businessNameToSubmitToTaxApi,
+        encryptedTaxId: encryptedTaxId as string,
+      });
+
+      updateQueueProfileData({
+        updateQueue,
+        userDataToSet,
+        tempProfileDataTaxId: tempProfileData.taxId,
+        encryptedTaxId,
+      });
+
+      if (getCurrentBusiness(userDataToSet).taxFilingData.state === "SUCCESS") {
+        if (displayBusinessName(business)) {
+          updateQueue.queueProfileData({
+            businessName: tempProfileData.businessName,
+          });
+        }
+
+        if (displayResponsibleOwnerName(business)) {
+          updateQueue.queueProfileData({
+            responsibleOwnerName: tempProfileData.responsibleOwnerName,
+          });
+        }
+      }
+
+      await updateQueue.update();
+    } catch {
+      return;
+    }
+
+    const { taxFilingData } = updateQueue.currentBusiness();
+
+    if (taxFilingData.state === "SUCCESS") {
+      analytics.event.tax_calendar_modal.submit.tax_deadlines_added_to_calendar();
+      onSuccess();
+      queueUpdateTaskProgress("determine-naics-code", "COMPLETED");
+      updateQueue.update();
+    }
+
+    if (taxFilingData.state === "PENDING") {
+      analytics.event.tax_calendar_modal.submit.business_exists_but_not_in_Gov2Go();
+      close();
+    }
+
+    if (taxFilingData.state === "FAILED") {
+      analytics.event.tax_calendar_modal.submit.tax_calendar_business_does_not_exist();
+    }
+
```

Comment:
[boulder] This whole block of code is almost identical to what's in `TaxAccessStepTwo`. I don't think I realized this before looking at this a second time. There's a good opportunity to do this DRY.